boundary_layer/util.py

```diff
@@ -21,6 +21,16 @@
 from boundary_layer.containers import ExecutionContext
 
 
+class GenericShell:
```

Comment:
This is an interesting addition.  I see what it's used for now, but it wasn't obvious to me at first.  Would you mind giving it a more descriptive name, maybe something like `GenericClassConstructor` or `GenericNamedParameterPasser`, or something like that?  Also perhaps a docstring with an example would help clear it up (I know we've not been great about writing docstrings in this project, but it's never too late to start).